### PR TITLE
Submission of py-rawpy

### PR DESCRIPTION
**Pull request by jsalort**
#### Description

New port for rawpy (Python wrapper for the LibRaw library).

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6 21G115 arm64
Xcode 14.0.1 14A400

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] checked your Portfile with `port lint --nitpick`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:13]: why do you need this?

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:12]: this isn't the correct usage of the `github.setup` line, the `v` version prefix should be after the actual version number as in:
```github.setup        letmaik rawpy 0.17.2 v```

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:17]: please add an obfuscated e-mail address as well

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:24]: this is set already by the `github` PG so please remove this line

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:26]: the patchfile should be set within the `if` block below as it doesn't need to happen for the `py-` stubfile

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:34]: are these dependency types correct? For example, is `cython` also used at runtime?

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:41]: this doesn't appear to work as the CI fails to find the headers

File: python/py-rawpy/files/setup.py.diff
Comment (reneeotten) [inline python/py-rawpy/files/setup.py.diff:10]: why would this need to be false?

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks for the submission @jsalort - a few comments and since the CI fails to build has of course to be resolved before this can be merged.

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile:13]: You mean downloading from github ? There is no source on Pypi, only pre-built wheels.

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile:12]: I thought the preference for MacPorts was be build everything ourselves. Besides, for the Mac version, they include their own version of libraw (which is also the reason for below).

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile:17]: OK, I will have a look at how to do that.

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile:24]: OK, will do.

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile:34]: I added all the dependencies that the project claims. It is possible however that cython is only used for building. I never tried to build a project with Cython and then run it without it. But if this is supposed to work, then I will move those dependencies to build only.

File: python/py-rawpy/files/setup.py.diff
Comment (jsalort) [reply python/py-rawpy/files/setup.py.diff:10]: Because otherwise they build their own libraw. I plan to contact upstream and propose to add a flag instead, but it will be easier to argue that it is useful if I can show them that MacPorts is a viable alternative for the installation.

Comment (jsalort) [reply python/py-rawpy/files/setup.py.diff:10]: To be clearer, they assume that linux has a libraw package available, hence the difference in behavior. Their conditionnals are ill-named. Mac=True implies to build libraw and change how to link to it. Mac=Windows=False implies Linux to them, and therefore normal building and linking process, which is what we want.

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile]: I don't understand why it is different on the CI and locally on my machine.

On the CI, there is
```bash
/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -arch x86_64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -D_HAS_LIBRAW_CONFIG_H=0 -Irawpy -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/numpy/core/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c rawpy/_rawpy.cpp -o build/temp.macosx-11.0-x86_64-cpython-310/rawpy/_rawpy.o
```

On my machine, there is
```bash
/usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -arch arm64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -D_HAS_LIBRAW_CONFIG_H=0 -Irawpy -I/opt/local/include/libraw -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/numpy/core/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c rawpy/_rawpy.cpp -o build/temp.macosx-12.0-arm64-cpython-310/rawpy/_rawpy.o
```

So `-I/opt/local/include/libraw` is correctly added on my machine, but not on the CI.
I tried using cppflags-append instead of cflags-append and cxx-flags-append, as was suggested by Joshua Root for the libraw port. It makes no difference (it works on my machine and not on the CI). I don't suppose that my machine being arm64 makes any difference in this matter?

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile]: OK.

File: python/py-rawpy/files/setup.py.diff
Comment (reneeotten) [reply python/py-rawpy/files/setup.py.diff:10]: okay - thanks! I must admit that I didn't look any further than the lines in the patchfile to see what's going on - it just seemed a little strange to me ;)

File: python/py-rawpy/Portfile
Comment (jmroot) [reply python/py-rawpy/Portfile]: You don't have a configure phase with the python portgroup, so `configure.*` options aren't used. It might be easiest to patch the needed flags into the `setup.py`.

Comment (jsalort) [reply python/py-rawpy/Portfile]: Ah, I wasn't aware that configure.* options were useless here. With that in mind, I figured that the reason it was working locally is because pkgconfig is installed. Indeed, the setup.py script uses pkgconfig to add CFLAGS. I tried in a virtual machine with blank MacPorts installation and I checked that adding pkgconfig as a build dependency fixes the issue.

However, the CI still fails, but now very early on, after 28 seconds, because /tmp/mpbb cannot be found. Is there some unrelated problems with the CI ?

Comment (jmroot) [reply python/py-rawpy/Portfile]: Yes, there was an issue with CI which has since been resolved.

File: python/py-rawpy/Portfile
Comment (jmroot) [reply python/py-rawpy/Portfile:21]: If you really don't want to list an email address, you can leave it as just a GitHub handle. It's just nice to be able to contact maintainers via email as well.

File: python/py-rawpy/Portfile
Comment (jmroot) [inline python/py-rawpy/Portfile:19]: Please set `python.pep517 yes` if that doesn't break the build.

Comment (jsalort) [reply python/py-rawpy/Portfile:19]: Just tried to enable it and it works.

File: python/py-rawpy/Portfile
Comment (jsalort) [reply python/py-rawpy/Portfile]: OK, I reconsidered, and added a dedicated email address.

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:43]: the `pytest` can be removed here as the default with `test.run` (when using the `python` PG) is to set `python.test_framework pytest`

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:13]: please move these lines up above the `name` field. You can then remove the separate `version` line

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:30]: no need for a newline here and before the closing bracket of the `if`-statement

File: python/py-rawpy/Portfile
Comment (reneeotten) [inline python/py-rawpy/Portfile:12]: latest version now is 0.17.3 - sorry, took too long to get this merged....

**Review by reneeotten — CHANGES_REQUESTED** [review]
@jsalort sorry for the delay, I did just take another look. If you'd be willing to change the last few things (including the update to the latest upstream version) then this should be good to go!

**Pull request merged**
